utils/biospecimen.js

```diff
@@ -653,6 +653,55 @@ const biospecimenAPIs = async (req, res) => {
         return res.status(200).json({data: response, code:200})
     }
 
+    else if(api == 'assignKit'){
+        if(req.method !== 'POST') {
+            return res.status(405).json(getResponseJSON('Only POST requests are accepted!', 405));
+        }
+        let requestData = req.body;
+        if(Object.keys(requestData).length === 0 ) return res.status(400).json(getResponseJSON('Request body is empty!', 400));
+        const { assignKitToParticipant } = require('./firestore');
+        const response = await assignKitToParticipant(requestData);
+        if(!response) return res.status(500).json({data: response, code:500});
```

Comment:
With the try catch wrapping the function returning an error object this code will not catch an actual processing error because the error object will evaluate truthy.  This can be handled a few ways one would be to seperate the 404 error and the 500 error or if we don't want that we can just modify the if and return a 500 for both senarios.

I would suggest either:
```
if(!response) {
    return res.status(404).json({data: response, code:404});
} else if (response instanceof Error) {
   return res.status(500).json({data: response, code: 500});
} else {
    return res.status(200).json({data: response, code:200});
}
```
or

```
 if (!response || response instanceof Error) {
   return res.status(500).json({data: response, code: 500});
} else {
    return res.status(200).json({data: response, code:200});
}